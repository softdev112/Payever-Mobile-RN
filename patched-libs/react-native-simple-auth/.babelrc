{
  "plugins": [
    "transform-es2015-modules-commonjs",
    "ramda-patch",
    "transform-object-rest-spread"
  ]
}
