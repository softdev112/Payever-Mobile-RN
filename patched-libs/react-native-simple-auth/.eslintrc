{
  "extends": "airbnb",
  "env": {
    "browser": true,
    "node": true,
    "mocha": true
  },
  "plugins": [
    "import"
  ],
  "rules": {
    "max-len": [2, 80, 2],
    "import/extensions": [0]
  }
}
