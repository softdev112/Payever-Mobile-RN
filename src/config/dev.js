/* eslint-disable max-len */
import local from './local';

export default Object.assign({
  siteUrl: 'https://mein.payever.de',
  api: {
    baseUrl: 'https://mein.payever.de',
    clientId: '1633_3vi6g4uiwmyow8044o4w0wo4s88cogosw84kw888kw408wok8c',
    clientSecret: '63nqx0kppzocw4sswok8gg800sk4w4kko4k0oo8gs4ow00wgos',
  },
  debug: {
    logApiCall: true,
  },

  oauthData: {
    twitter: {
      appId: 'oRCkHDAnAFSXiO1wayGlpBkwn',
      appSecret: 'C3FUgIBnjdn1Je6dD5NX5AboFEJ1vWqVVwOLJ03Itk0Xp58but',
      callback: '/social/auth/twitter',
    },

    facebook: {
      appId: '438908279593357',
      appSecret: 'cd96a62e92a5bc498896209613e2d0bf',
      callback: '/social/auth/facebook',
      scope: 'email public_profile',
      fields: [
        'id', 'name', 'first_name', 'last_name',
        'picture.type(large)', 'email', 'cover', 'birthday',
      ],
    },

    google: {
      appId: '463919467171-u54vacd9oiso7l14f97g38odsak91njg.apps.googleusercontent.com',
      appSecret: 'SihRvMNWT2j5088wEZ_A2MRR',
      callback: 'com.googleusercontent.apps.463919467171-u54vacd9oiso7l14f97g38odsak91njg:/google',
      scope: 'https://www.googleapis.com/auth/content',
    },

    linkedIn: {
      appId: '78a1v96xm7x1tt',
      appSecret: 'Umu6R25x1w7NiDt6',
      callback: 'https://localhost:3003/auth/linkedin',
      scope: 'r_basicprofile r_emailaddress',
    },
  },
}, local);