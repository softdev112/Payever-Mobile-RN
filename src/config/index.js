import generatedConfig from './_autogenerated';

export class Config {
  siteUrl: string;
  api: {
    baseUrl: string;
    clientId: string;
    clientSecret: string;
  };
  debug: {
    consoleLevel: boolean;
    logApiCall: boolean;
    logMobx: boolean;
    logSLevel: boolean;
    logSUrl: string;
    logWampCall: boolean;
  };

  constructor(data) {
    data.debug = data.debug || {};
    Object.assign(this, data);
  }
}

export default new Config(generatedConfig);