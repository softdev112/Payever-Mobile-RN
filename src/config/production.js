/* eslint-disable max-len */
export default {
  siteUrl: 'https://mein.payever.de',
  // It's better to prevent using clientSecret. But after analyzing a lot
  // of resources I've made a decision that is's impossible.
  // FOSOAuthServerBundle doesn't support password auth without client_secret
  // https://github.com/FriendsOfSymfony/FOSOAuthServerBundle/issues/115
  api: {
    baseUrl: 'https://mein.payever.de',
    clientId: '1633_3vi6g4uiwmyow8044o4w0wo4s88cogosw84kw888kw408wok8c',
    clientSecret: '63nqx0kppzocw4sswok8gg800sk4w4kko4k0oo8gs4ow00wgos',
  },
  reqTimeout: 10000,

  oauthData: {
    twitter: {
      appId: 'oRCkHDAnAFSXiO1wayGlpBkwn',
      appSecret: 'C3FUgIBnjdn1Je6dD5NX5AboFEJ1vWqVVwOLJ03Itk0Xp58but',
      callback: '/social/auth/twitter',
    },

    facebook: {
      appId: '302838276848223',
      appSecret: '188dcf1590b667929d5f8b4538b00075',
      callback: 'fb302838276848223://authorize',
      scope: 'email public_profile',
      fields: [
        'id', 'name', 'first_name', 'last_name',
        'picture.type(large)', 'email', 'cover', 'birthday',
      ],
    },

    google: {
      appId: '463919467171-u54vacd9oiso7l14f97g38odsak91njg.apps.googleusercontent.com',
      appSecret: 'SihRvMNWT2j5088wEZ_A2MRR',
      callback: 'com.googleusercontent.apps.463919467171-u54vacd9oiso7l14f97g38odsak91njg:/google',
      scope: 'https://www.googleapis.com/auth/content',
    },

    linkedIn: {
      appId: '78a1v96xm7x1tt',
      appSecret: 'Umu6R25x1w7NiDt6',
      callback: 'https://localhost:3003/auth/linkedin',
      scope: 'r_basicprofile r_emailaddress',
    },
  },
};