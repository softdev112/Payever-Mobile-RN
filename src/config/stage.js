export default {
  siteUrl: 'https://stage.payever.de',
  api: {
    baseUrl: 'https://stage.payever.de',
    clientId: '117_5sd54ii2lyo8sg84g0w4w08wkw4040gks4k0sk0wgc44w48404',
    clientSecret: '4jsu5lynawsg8g4cs80w4kcsocwcggwo048ogs8c844gk80w80'
  }
}