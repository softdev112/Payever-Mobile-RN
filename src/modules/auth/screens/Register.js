import { Component } from 'react';
import { inject, observer } from 'mobx-react/native';
import type { Navigator } from 'react-native-navigation';
import { google, facebook, linkedIn, twitter } from 'react-native-simple-auth';
import { SocialIconButton, NavBar, StyleSheet, TextButton, View } from 'ui';
import { log } from 'utils';
import { showScreen } from '../../../common/Navigation';
import type { Config } from '../../../config';
import type AuthStore from '../../../store/auth';

@inject('auth', 'config')
@observer
export default class Register extends Component {
  static navigatorStyle = {
    navBarHidden: true,
  };

  props: {
    config: Config;
    auth: AuthStore;
    navigator: Navigator;
  };

  onRegisterWithTwitter() {
    const { auth, config: { siteUrl, oauthData } } = this.props;
    twitter({
      appId: oauthData.twitter.appId,
      appSecret: oauthData.twitter.appSecret,
      callback: siteUrl + oauthData.twitter.callback,
    }).then(async ({ credentials }) => {
      await auth.signInWithSocial('twitter', credentials);

      const isLoggedIn = await auth.checkAuth();
      if (isLoggedIn) {
        showScreen('dashboard.ChooseAccount');
      }
    }).catch(log.error);
  }

  onRegisterWithFacebook() {
    const { auth, config: { oauthData } } = this.props;

    facebook({
      appId: oauthData.facebook.appId,
      appSecret: oauthData.facebook.appSecret,
      callback: oauthData.facebook.callback,
      scope: oauthData.facebook.scope,
      fields: oauthData.facebook.fields,
    }).then(async ({ credentials }) => {
      await auth.signInWithSocial('facebook', credentials);

      const isLoggedIn = await auth.checkAuth();
      if (isLoggedIn) {
        showScreen('dashboard.ChooseAccount');
      }
    }).catch(log.error);
  }

  onRegisterWithGoogle() {
    const { auth, config: { oauthData } } = this.props;

    google({
      appId: oauthData.google.appId,
      appSecret: oauthData.google.appSecret,
      callback: oauthData.google.callback,
      scope: oauthData.google.scope,
    }).then(async ({ credentials }) => {
      await auth.signInWithSocial('google', credentials);

      const isLoggedIn = await auth.checkAuth();
      if (isLoggedIn) {
        showScreen('dashboard.ChooseAccount');
      }
    }).catch(log.error);
  }

  onRegisterWithLinkedIn() {
    const { auth, config: { oauthData } } = this.props;

    linkedIn({
      appId: oauthData.linkedIn.appId,
      appSecret: oauthData.linkedIn.appSecret,
      callback: oauthData.linkedIn.callback,
      scope: oauthData.linkedIn.scope,
    }).then(async ({ credentials }) => {
      await auth.signInWithSocial('linkedin', credentials);

      const isLoggedIn = await auth.checkAuth();
      if (isLoggedIn) {
        showScreen('dashboard.ChooseAccount');
      }
    }).catch(log.error);
  }

  onRegisterWithEmail() {
    const { navigator } = this.props;
    navigator.push({ screen: 'auth.RegisterWithEmail', animated: true });
  }

  render() {
    return (
      <View style={styles.container}>
        <NavBar>
          <NavBar.Back title="Back" />
          <NavBar.Title title="Add a contact" />
        </NavBar>

        <View style={styles.content}>
          <SocialIconButton
            style={styles.facebookBtn}
            title="REGISTER WITH FACEBOOK"
            iconSource="fa-facebook"
            onPress={::this.onRegisterWithFacebook}
          />

          <SocialIconButton
            style={styles.googleBtn}
            title="REGISTER WITH GOOGLE"
            iconSource="fa-google"
            onPress={::this.onRegisterWithGoogle}
          />

          <SocialIconButton
            style={styles.twitterBtn}
            title="REGISTER WITH TWITTER"
            iconSource="fa-twitter"
            onPress={::this.onRegisterWithTwitter}
          />

          <SocialIconButton
            style={styles.linkedInBtn}
            title="REGISTER WITH LINKEDIN"
            iconSource="fa-linkedin"
            onPress={::this.onRegisterWithLinkedIn}
          />

          <TextButton
            style={styles.withEmailBtn}
            title="REGISTER WITH EMAIL"
            onPress={::this.onRegisterWithEmail}
          />
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },

  content: {
    alignItems: 'center',
    marginTop: 50,
    flex: 1,
  },

  facebookBtn: {
    backgroundColor: '$pe_color_facebook',
  },

  googleBtn: {
    marginTop: 16,
    backgroundColor: '$pe_color_google',
  },

  twitterBtn: {
    backgroundColor: '$pe_color_twitter',
    marginTop: 16,
  },

  linkedInBtn: {
    marginTop: 16,
    backgroundColor: '$pe_color_linkedIn',
  },

  withEmailBtn: {
    marginTop: 16,
    alignSelf: 'center',
  },
});