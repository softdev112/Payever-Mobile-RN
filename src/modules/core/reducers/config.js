import { Map } from 'immutable';
import config from '../../../config';

const initialState = Map(config);

/**
 * Share config through store
 */
export default function configReducer(state = initialState) {
  return state;
}