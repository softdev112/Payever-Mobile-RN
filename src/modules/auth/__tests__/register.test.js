/* eslint-disable max-len */
import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { google, facebook, linkedIn, twitter } from 'react-native-simple-auth';
import Register from '../screens/Register';
import Store from '../../../store';
import config from '../../../config';

jest.mock('../../../common/Navigation')
  .mock('react-native-simple-auth', () => ({
    twitter: jest.fn(() => Promise.resolve(1)),
    google: jest.fn(() => Promise.resolve(1)),
    facebook: jest.fn(() => Promise.resolve(1)),
    linkedIn: jest.fn(() => Promise.resolve(1)),
  }));

const navigator = {
  setOnNavigatorEvent: jest.fn(),
  resetTo: jest.fn(),
  push: jest.fn(),
};

describe('modules/auth/screens/Register', () => {
  let store;
  let auth;

  beforeAll(() => {
    Register.prototype.getChildContext = function getContext() {
      return { navigator };
    };

    Register.childContextTypes = {
      navigator: React.PropTypes.object,
    };
  });

  beforeEach(() => {
    store = new Store(config);
    auth = store.auth;
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('Register should render correctly', () => {
    const tree = renderer.create(
      <Register
        auth={auth}
        config={store.config}
        navigator={navigator}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Register/onRegisterWithEmail() should call navigator.push to RegisterWithEmail screen', () => {
    const wrapper = shallow(
      <Register
        auth={auth}
        config={store.config}
        navigator={navigator}
      />
    );

    wrapper.dive()
      .find('TextButton')
      .at(0)
      .props()
      .onPress();

    expect(navigator.push).toHaveBeenCalled();
    expect(navigator.push).toHaveBeenCalledWith(
      { animated: true, screen: 'auth.RegisterWithEmail' }
    );
  });

  it('Register/onRegisterWithLinkedIn() should call register with linkedIn', () => {
    const wrapper = shallow(
      <Register
        auth={auth}
        config={store.config}
        navigator={navigator}
      />
    );

    wrapper.dive()
      .find('SocialIconButton')
      .at(3)
      .props()
      .onPress();

    expect(linkedIn).toHaveBeenCalled();
  });

  it('Register/onRegisterWithTwitter() should call register with twitter', () => {
    const wrapper = shallow(
      <Register
        auth={auth}
        config={store.config}
        navigator={navigator}
      />
    );

    wrapper.dive()
      .find('SocialIconButton')
      .at(2)
      .props()
      .onPress();

    expect(twitter).toHaveBeenCalled();
  });

  it('Register/onRegisterWithGoogle() should call register with google', () => {
    const wrapper = shallow(
      <Register
        auth={auth}
        config={store.config}
        navigator={navigator}
      />
    );

    wrapper.dive()
      .find('SocialIconButton')
      .at(1)
      .props()
      .onPress();

    expect(google).toHaveBeenCalled();
  });

  it('Register/onRegisterWithFacebook() should call register with facebook', () => {
    const wrapper = shallow(
      <Register
        auth={auth}
        config={store.config}
        navigator={navigator}
      />
    );

    wrapper.dive()
      .find('SocialIconButton')
      .at(0)
      .props()
      .onPress();

    expect(facebook).toHaveBeenCalled();
  });
});