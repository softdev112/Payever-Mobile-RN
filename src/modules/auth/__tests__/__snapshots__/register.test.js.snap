// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`modules/auth/screens/Register Register should render correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFF",
            "borderBottomColor": "#e1e1e1",
            "borderBottomWidth": 1,
            "elevation": 8,
            "flexDirection": "row",
            "height": 55,
            "justifyContent": "flex-end",
            "marginTop": 15,
            "paddingHorizontal": 18,
            "zIndex": 1000,
          },
        ],
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "bottom": 0,
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "left": 18,
            "position": "absolute",
            "top": 0,
            "zIndex": 1002,
          },
          null,
        ]
      }
    >
      <View
        accessibilityComponentType="button"
        accessibilityLabel={undefined}
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        accessible={true}
        hitSlop={
          Object {
            "bottom": 10,
            "left": 10,
            "right": 10,
            "top": 10,
          }
        }
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "flexDirection": "row",
            "justifyContent": "flex-end",
            "opacity": 1,
            "padding": 6,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          source="icon-arrow-left-ios-24"
          style={
            Array [
              Object {
                "fontFamily": "payeverIcons",
                "fontSize": 24,
              },
              Array [
                Object {
                  "fontFamily": "payeverIcons",
                  "fontSize": 24,
                },
                Array [
                  Object {
                    "height": 24,
                    "width": 24,
                  },
                  Object {
                    "color": "#0084ff",
                  },
                  undefined,
                ],
              ],
            ]
          }
          touchStyle={
            Object {
              "alignItems": "center",
              "borderColor": "blue",
              "borderWidth": 1,
              "justifyContent": "center",
            }
          }
        >
          
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#0084ff",
                "fontSize": 16,
                "marginLeft": 2,
              },
              Object {
                "fontSize": 16,
                "fontWeight": "400",
                "marginLeft": 2,
              },
            ]
          }
        >
          Back
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "bottom": 0,
          "flexDirection": "row",
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "zIndex": 1001,
        }
      }
    >
      <View
        accessibilityComponentType="button"
        accessibilityLabel={undefined}
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        accessible={true}
        hitSlop={
          Object {
            "bottom": 10,
            "left": 10,
            "right": 10,
            "top": 10,
          }
        }
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "flexDirection": "row",
            "justifyContent": "center",
            "opacity": 1,
            "padding": 6,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#000",
                "fontSize": 18,
                "textAlign": "center",
              },
              Object {},
            ]
          }
        >
          Add a contact
        </Text>
      </View>
    </View>
    <Image
      source={1}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": null,
          "justifyContent": "flex-end",
          "paddingLeft": 20,
          "resizeMode": "stretch",
          "width": null,
          "zIndex": 1002,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "marginTop": 50,
      }
    }
  >
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#45619d",
          "borderColor": "#45619d",
          "borderRadius": 25,
          "borderWidth": 1,
          "height": 40,
          "justifyContent": "center",
          "opacity": 1,
          "paddingHorizontal": 10,
          "width": 260,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          source="fa-facebook"
          style={
            Array [
              Object {
                "color": "#b5b9be",
                "fontFamily": "FontAwesome",
                "fontSize": 16,
              },
              Array [
                Object {
                  "color": "#b5b9be",
                  "fontFamily": "FontAwesome",
                  "fontSize": 16,
                },
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 25,
                    "marginRight": 8,
                  },
                  undefined,
                ],
              ],
            ]
          }
        >
          
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 16,
                "fontWeight": "400",
                "textAlign": "center",
              },
              undefined,
            ]
          }
        >
          REGISTER WITH FACEBOOK
        </Text>
      </View>
    </View>
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#CC4A3C",
          "borderColor": "#CC4A3C",
          "borderRadius": 25,
          "borderWidth": 1,
          "height": 40,
          "justifyContent": "center",
          "marginTop": 16,
          "opacity": 1,
          "paddingHorizontal": 10,
          "width": 260,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          source="fa-google"
          style={
            Array [
              Object {
                "color": "#b5b9be",
                "fontFamily": "FontAwesome",
                "fontSize": 16,
              },
              Array [
                Object {
                  "color": "#b5b9be",
                  "fontFamily": "FontAwesome",
                  "fontSize": 16,
                },
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 25,
                    "marginRight": 8,
                  },
                  undefined,
                ],
              ],
            ]
          }
        >
          
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 16,
                "fontWeight": "400",
                "textAlign": "center",
              },
              undefined,
            ]
          }
        >
          REGISTER WITH GOOGLE
        </Text>
      </View>
    </View>
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#50abf1",
          "borderColor": "#50abf1",
          "borderRadius": 25,
          "borderWidth": 1,
          "height": 40,
          "justifyContent": "center",
          "marginTop": 16,
          "opacity": 1,
          "paddingHorizontal": 10,
          "width": 260,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          source="fa-twitter"
          style={
            Array [
              Object {
                "color": "#b5b9be",
                "fontFamily": "FontAwesome",
                "fontSize": 16,
              },
              Array [
                Object {
                  "color": "#b5b9be",
                  "fontFamily": "FontAwesome",
                  "fontSize": 16,
                },
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 25,
                    "marginRight": 8,
                  },
                  undefined,
                ],
              ],
            ]
          }
        >
          
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 16,
                "fontWeight": "400",
                "textAlign": "center",
              },
              undefined,
            ]
          }
        >
          REGISTER WITH TWITTER
        </Text>
      </View>
    </View>
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#0077b5",
          "borderColor": "#0077b5",
          "borderRadius": 25,
          "borderWidth": 1,
          "height": 40,
          "justifyContent": "center",
          "marginTop": 16,
          "opacity": 1,
          "paddingHorizontal": 10,
          "width": 260,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          source="fa-linkedin"
          style={
            Array [
              Object {
                "color": "#b5b9be",
                "fontFamily": "FontAwesome",
                "fontSize": 16,
              },
              Array [
                Object {
                  "color": "#b5b9be",
                  "fontFamily": "FontAwesome",
                  "fontSize": 16,
                },
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 25,
                    "marginRight": 8,
                  },
                  undefined,
                ],
              ],
            ]
          }
        >
          
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 16,
                "fontWeight": "400",
                "textAlign": "center",
              },
              undefined,
            ]
          }
        >
          REGISTER WITH LINKEDIN
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignSelf": "flex-start",
          },
          Object {
            "alignSelf": "center",
            "marginTop": 16,
          },
        ]
      }
    >
      <View
        accessibilityComponentType="button"
        accessibilityLabel={undefined}
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#0084ff",
                "fontSize": 16,
                "fontWeight": "400",
              },
              undefined,
            ]
          }
        >
          REGISTER WITH EMAIL
        </Text>
      </View>
    </View>
  </View>
</View>
`;
